### PR TITLE
Marian distill scripts + integration test

### LINKED ISSUE #6797: 2 Slow Test Failures That Sam Can Fix
Both related to fp16

1)
```
        if torch.cuda.is_available():
            testargs += ["--fp16", "--gpus=1"]
    
        with patch.object(sys, "argv", testargs):
>           result = run_pl_glue.main()
```

2)
```
examples/seq2seq/test_bash_script.py::test_train_mbart_cc25_enro_script
```
needs rerun/ cannot use fp16

### PR DESCRIPTION
**Pull request by sshleifer**
- Fix 2 integration tests in examples
- Add new integration test for marian distillation script
- add 2 marian distillation scripts (but dont document them, will do that once they are slightly better).

### CODE REVIEW

File: examples/seq2seq/test_bash_script.py [hunk 1]
```diff
@@ -35,34 +37,29 @@ def test_model_download():
...
     }
 
     # Clean up bash script
     bash_script = Path("examples/seq2seq/train_mbart_cc25_enro.sh").open().read().split("finetune.py")[1].strip()
-    bash_script = bash_script.replace("\\\n", "").strip().replace("$@", "")
+    bash_script = bash_script.replace("\\\n", "").strip().replace('"$@"', "")
     for k, v in env_vars_to_replace.items():
         bash_script = bash_script.replace(k, str(v))
-    output_dir = tempfile.mkdtemp(prefix="output")
+    output_dir = tempfile.mkdtemp(prefix="output_mar")
```
Comment (sshleifer) [inline examples/seq2seq/test_bash_script.py:54]:
```suggestion
    output_dir = tempfile.mkdtemp(prefix="output_mbart")
```

File: examples/test_examples.py [hunk 2]
```diff
@@ -114,7 +114,9 @@ def test_run_pl_glue(self):
             --max_seq_length=128
             """.split()
         if torch.cuda.is_available():
-            testargs += ["--fp16", "--gpus=1"]
+            testargs += ["--gpus=1"]
+        if is_cuda_and_apex_avaliable():
```
Comment (sgugger) [inline examples/test_examples.py:118]: Typo here I think?

Comment (sshleifer) [reply examples/test_examples.py:118]: I only want fp16 if apex available and cuda available. Am I missing something?

Comment (sgugger) [reply examples/test_examples.py:118]:
```suggestion
        if is_cuda_and_apex_available():
```
(GitHub didn't let me make a suggestion earlier for some reason.

Comment (sshleifer) [reply examples/test_examples.py:118]: incredible catch!

**Pull request merged**